src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private static final int RESIZE_NUMBER = 2;
+    private int size;
+    private Node<K, V>[] table = new Node[DEFAULT_CAPACITY];
+    private int capacity = DEFAULT_CAPACITY;
+    private int threshould = (int) (capacity * DEFAULT_LOAD_FACTOR);
+
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshould) {
+            resizeMap();
+        }
+        int hash = hashCalculator(key);
+        int bucket = hash % capacity;
```

Comment:
combine into one method, getIndexFromKey

Reply:
i need hash in my node

Reply:
![Uploading Screenshot_3.jpg…]()